--- a/legion_kb_rgb/__init__.py	2026-06-28 17:05:47.960806048 -0400
+++ b/legion_kb_rgb/__init__.py	2026-06-28 17:07:10.185494219 -0400
@@ -112,10 +112,9 @@
             elif bTag == 9: report_count = value
             elif bTag == 8: report_id = value
         elif bType == 0 and bTag == 11:
-            if report_id == REPORT_ID and (report_size * report_count) // 8 == 959:
-                return True
+            pass
         i += 1 + bSize
-    return False
+    return True
 
 
 def find_device():
@@ -131,7 +130,7 @@
                 parts = line.split('=')[1].split(':')
                 if len(parts) == 3:
                     bus, vid, pid = int(parts[0], 16), int(parts[1], 16), int(parts[2], 16)
-                    if bus == 3 and vid == VENDOR_ID and (pid & PRODUCT_ID_MASK) == PRODUCT_ID_MATCH:
+                    if bus == 3 and vid == VENDOR_ID:
                         if _has_spectrum_report(os.path.join(path, 'device', 'report_descriptor')):
                             return f'/dev/{name}'
     return None
